### ISSUE TITLE
Add URL sanitiser

### ISSUE THREAD

**Issue by MuntashirAkon** [MuntashirAkon/AppManager#256]
Interceptor can be used to sanitise http(s) URLs.

**Comment by Atrate** [comment]
It may also be used to rewrite some URLs to more privacy-respecting sites (e.g. reddit -> teddit) like https://f-droid.org/repo/app.fedilab.nitterizemelite

**Comment by MuntashirAkon** [comment]
Yeah, I will allow some custom rules. These could be supplied as built-in, I suppose.

**Comment by UltraBlackLinux** [comment]
@Atrate 
(Teddit.com best website)

**Comment by MuntashirAkon** [comment]
AM will only allow simple rules. Privacy redirect seems to be a complex feature. So, here's what I'm going to do:

We will create an extension (simply copying the source code from UntrackMe) which will set the redirection rules in AM. This way people can configure the instances, etc. without having to leave the app.

But before that, I need to implement #75 (MIME and Schema based default apps).

**Comment by MuntashirAkon** [comment]
We shall follow URLChecker and privacy-redirect apps to implement the features.

**Comment by shuvashish76** [comment]
> We will create an extension (simply copying the source code from UntrackMe) which will set the redirection rules in AM. This way people can configure the instances, etc. without having to leave the app.

For example URLChecker has [this 3rd party module](https://github.com/Raj9039852537/urlChecker-patternchecker-config) ([discussion](https://github.com/TrianguloY/UrlChecker/discussions/122)). We can have such extension/module similar to [community based Exported rules, Profiles](https://github.com/MuntashirAkon/AppManager/discussions/503). Add proper instruction about how to create such extension & share with AM community. And for general users, instruction on how to take advantage of community created extensions.

Also have a look on [ClearURLs Kotlin](https://github.com/1fexd/clearurlkt)  (removes tracking parameters), [FastFirward Kotlin](https://github.com/1fexd/fastforwardkt) (extracts redirects from url parameters), [LibRedirect Kotlin](https://github.com/1fexd/libredirectkt) (redirects to non-proprietary frontends) which are used on [LinkSheet](https://github.com/LinkSheet/LinkSheet) app.
